crates/lib/src/lib.rs

```diff
@@ -205,52 +205,54 @@ pub enum IndexType {
     Hash,
 }
 
-#[derive(Debug, Default, Clone, Copy, Eq, PartialEq, PartialOrd, Ord, de::Deserialize, ser::Serialize)]
-pub enum ColumnIndexAttribute {
-    #[default]
-    UnSet = 0,
-    /// Unique + AutoInc
-    Identity = 1,
-    /// Index unique
-    Unique = 2,
-    ///  Index no unique
-    Indexed = 3,
-    /// Generate the next [Sequence]
-    AutoInc = 4,
-    /// Primary key column (implies Unique)
-    PrimaryKey = 5,
-    /// PrimaryKey + AutoInc
-    PrimaryKeyAuto = 6,
+bitflags::bitflags! {
+    #[derive(Debug, Default, Clone, Copy, Eq, PartialEq, PartialOrd, Ord)]
+    pub struct ColumnIndexAttribute: u8 {
+        const UNSET = Self::empty().bits();
+        ///  Index no unique
+        const INDEXED = 0b0001;
+        /// Generate the next [Sequence]
+        const AUTO_INC = 0b0010;
+        /// Index unique
+        const UNIQUE = Self::INDEXED.bits() | 0b0100;
+        /// Unique + AutoInc
+        const IDENTITY = Self::UNIQUE.bits() | Self::AUTO_INC.bits();
+        /// Primary key column (implies Unique)
+        const PRIMARY_KEY = Self::UNIQUE.bits() | 0b1000;
+        /// PrimaryKey + AutoInc
+        const PRIMARY_KEY_AUTO = Self::PRIMARY_KEY.bits() | Self::AUTO_INC.bits();
+    }
 }
 
 impl ColumnIndexAttribute {
     pub const fn is_unique(self) -> bool {
-        matches!(
-            self,
-            Self::Identity | Self::Unique | Self::PrimaryKey | Self::PrimaryKeyAuto
-        )
+        self.contains(Self::UNIQUE)
     }
     pub const fn is_autoinc(self) -> bool {
-        matches!(self, Self::Identity | Self::AutoInc | Self::PrimaryKeyAuto)
+        self.contains(Self::AUTO_INC)
     }
     pub const fn is_primary(self) -> bool {
-        matches!(self, Self::PrimaryKey | Self::PrimaryKeyAuto)
+        self.contains(Self::PRIMARY_KEY)
     }
 }
 
 impl TryFrom<u8> for ColumnIndexAttribute {
     type Error = ();
 
     fn try_from(v: u8) -> Result<Self, Self::Error> {
-        match v {
```

Comment:
My only concern is here. The bitflags crate does not allow us to guarantee that only "valid" bits will be set. So I'm concerned that we are removing this code which will check for invalid bits early.

Reply:
`from_bits` specifically does verify that only known bits are set - it returns `Option` which I'm converting to a `Result` here.

Reply:
The top level docs confused me then: "guarantee only bits corresponding to defined flags will ever be set."

Could we add a comment about this then? Then this looks great to me.

Reply:
Oh yeah, bitflags-generated structure itself doesn't guarantee that - if you really want to, you can still set arbitrary bits via dedicated methods - but specific conversion methods like `from_bits` have those extra checks. I'll add a comment.

Reply:
OTOH... looking at it again, the error message in `ok_or_else` seems to serve as a good enough comment?